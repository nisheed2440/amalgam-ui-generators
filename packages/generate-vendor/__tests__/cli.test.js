const { exec } = require('child_process');
const path = require('path');
const cli = require('../lib/cli');
const generateVendor = require('../lib/generate-vendor');

// Mock the generator
jest.mock('../lib/generate-vendor', () => ({
  generateVendor: jest.fn(),
}));

describe('@amalgam-ui/generate-vendor cli', () => {
  it('should not have the parameters as expected', () => {
    const argv = cli(process.cwd()).parse(['']);
    expect(argv.vendorFolderName).toBe('');
    expect(argv.namespace).toBeUndefined();
    expect(argv.n).toBeUndefined();
    expect(argv.dest).toBe('packages/');
    expect(argv.d).toBe('packages/');
  });

  it('should have the parameters as expected', () => {
    const argv = cli(process.cwd()).parse([
      'Test Vendor',
      '--namespace',
      '@test',
      '--dest',
      'test',
    ]);
    expect(argv.vendorFolderName).toBe('Test Vendor');
    expect(argv.namespace).toBe('@test');
    expect(argv.n).toBe('@test');
    expect(argv.dest).toBe('test');
    expect(argv.d).toBe('test');
  });

  it('should show help info', done => {
    exec(`node ${path.resolve(__dirname, '../bin/generate-vendor')} --help`, (error, stdout) => {
      if (error) {
        throw error;
      }
      expect(stdout).toMatchSnapshot();
      done();
    });
  });

  it('should show version', done => {
    exec(`node ${path.resolve(__dirname, '../bin/generate-vendor')} --version`, (error, stdout) => {
      if (error) {
        throw error;
      }
      expect(stdout).toMatchSnapshot();
      done();
    });
  });

  it('should call the generateVendor fn', () => {
    const argv = cli(process.cwd()).parse(['Test Vendor', '--namespace', '@test']);
    expect(generateVendor.generateVendor).toHaveBeenCalledWith(argv);
  });
});
