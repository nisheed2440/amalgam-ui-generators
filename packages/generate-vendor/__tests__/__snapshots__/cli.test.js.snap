// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@amalgam-ui/generate-vendor cli should show help info 1`] = `
"generate-vendor <vendor-folder-name>

Create vendor folder for Amalgam UI framework.

Positionals:
  vendor-folder-name  The name of the vendor folder                     [string]

Options:
  -n, --namespace  The namespace for the vendor package.json
  -d, --dest       The destination path for the vendor files
                                                          [default: \\"packages/\\"]
  -h, --help       Show help                                           [boolean]
  -v, --version    Show version number                                 [boolean]
"
`;

exports[`@amalgam-ui/generate-vendor cli should show version 1`] = `
"0.0.1-alpha.0
"
`;
